src/Aspire.Cli/Commands/UpdateCommand.cs

```diff
@@ -0,0 +1,53 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.CommandLine;
+using Aspire.Cli.Configuration;
+using Aspire.Cli.Interaction;
+using Aspire.Cli.Packaging;
+using Aspire.Cli.Projects;
+using Aspire.Cli.Resources;
+using Aspire.Cli.Utils;
+
+namespace Aspire.Cli.Commands;
+
+internal sealed class UpdateCommand : BaseCommand
+{
+    private readonly IProjectLocator _projectLocator;
+    private readonly IInteractionService _interactionService;
+    private readonly IPackagingService _packagingService;
+    private readonly IProjectUpdater _projectUpdater;
+
+    public UpdateCommand(IProjectLocator projectLocator, IPackagingService packagingService, IProjectUpdater projectUpdater, IInteractionService interactionService, IFeatures features, ICliUpdateNotifier updateNotifier) : base("update", UpdateCommandStrings.Description, features, updateNotifier)
+    {
+        ArgumentNullException.ThrowIfNull(projectLocator);
+        ArgumentNullException.ThrowIfNull(interactionService);
+        ArgumentNullException.ThrowIfNull(packagingService);
+        ArgumentNullException.ThrowIfNull(projectUpdater);
+
+        _projectLocator = projectLocator;
+        _interactionService = interactionService;
+        _packagingService = packagingService;
+        _projectUpdater = projectUpdater;
+    }
+
+    protected override async Task<int> ExecuteAsync(ParseResult parseResult, CancellationToken cancellationToken)
+    {
+        try
+        {
+            var projectFile = await _projectLocator.UseOrFindAppHostProjectFileAsync(null, cancellationToken);
+            var channels = await _packagingService.GetChannelsAsync(cancellationToken);
+
+            var channel = await _interactionService.PromptForSelectionAsync("Select a channel:", channels, (c) => c.Name, cancellationToken);
+
+            await _projectUpdater.UpdateProjectAsync(projectFile!, channel, cancellationToken);
+        }
+        catch (ProjectUpdaterException ex)
+        {
+            _interactionService.DisplayError(ex.Message);
```

Comment:
Remind me, DisplayError escapes the input right?

Reply:
Explicitly added escapes for this just in case.